carball/analysis/stats/boost/boost.py

```diff
@@ -33,10 +41,34 @@ def get_player_boost_usage(player_dataframe: pd.DataFrame) -> np.float64:
 
     @staticmethod
     def get_player_boost_usage_max_speed(player_dataframe: pd.DataFrame) -> np.float64:
-        """TODO: do this"""
         _diff = -player_dataframe.boost.diff()
-        boost_usage = _diff[_diff > 0].sum() / 255 * 100
-        return boost_usage
+
+        speed: pd.Series = (player_dataframe.vel_x ** 2 +
+                            player_dataframe.vel_y ** 2 +
+                            player_dataframe.vel_z ** 2) ** 0.5
+
+        _diff = _diff.rename('boost')
+        speed = speed.rename('speed')
+
+        combined = pd.concat([_diff, speed], axis=1)
+
+        filtered_1 = combined[combined.speed > 22000]
+        filtered_2 = filtered_1[filtered_1.boost > 0]
+        filtered_3 = filtered_2[filtered_2.boost < 10]
+        boost_usage = filtered_3.boost.sum() / 255 * 100
+        return float(boost_usage)
+
+    @staticmethod
+    def get_time_with_max_boost(data_frame: pd.DataFrame, player_dataframe: pd.DataFrame) -> np.float64:
+        return sum_deltas_by_truthy_data(data_frame, player_dataframe.boost > 252.45)
```

Comment:
what's 252.45 from?
If it's stored as an int, wouldn't 252 be equivalent and less confusing

Reply:
boost is not an int.  visually it is ceiled.  This number captures the ceil portion.
So if it is visually at 100 this triggers.